### PR TITLE
LLAMA-2652 Device is not getting out of standby and when maintenance …

### PR DESCRIPTION
**Pull request by michaelamal**
…starts

Reason for change: remove  RequestReboot as it is not in use
Test Procedure: build procedure
Risks: No

Signed-off-by: Michael Amal Michael_AmalAnand@comcast.com

### CODE REVIEW

File: MaintenanceManager/MaintenanceManager.cpp [hunk 1]
```diff
@@ -813,6 +800,8 @@ namespace WPEFramework {
                     string abort_flag="";
 
                     /* only one maintenance at a time */
+                    /* Lock so that g_notify_status will not be updated  further */
+                    m_statusMutex.lock();
```
Comment (anand-ky) [inline MaintenanceManager/MaintenanceManager.cpp:804]: Why are we locking at 2 places? here and within onMaintenanceStatusChange() before updating g_notify_status. Once this lock is taken here, it's going to block the task_execution_thread from updating g_notify_status.

File: MaintenanceManager/MaintenanceManager.cpp [hunk 2]
```diff
@@ -585,23 +578,17 @@ namespace WPEFramework {
                             m_notify_status=MAINTENANCE_ERROR;
```
Comment (anand-ky) [inline MaintenanceManager/MaintenanceManager.cpp:578]: Pls change m_notify_status to notify_status to indicate it's a local variable. Also pls change g_notify_status to m_notify_status to indicate it's a member and not global.

**Review by anand-ky — CHANGES_REQUESTED** [review]

File: MaintenanceManager/MaintenanceManager.cpp (hunk 2 above)
Comment (michaelamal) [reply MaintenanceManager/MaintenanceManager.cpp]: fixed

File: MaintenanceManager/MaintenanceManager.cpp [hunk 3]
```diff
@@ -550,7 +553,7 @@ namespace WPEFramework {
                 if ( (g_task_status & TASKS_COMPLETED ) == TASKS_COMPLETED ){
                     if ( (g_task_status & ALL_TASKS_SUCCESS) == ALL_TASKS_SUCCESS ){ // all tasks success
                         LOGINFO("DBG:Maintenance Successfully Completed!!");
-                        m_notify_status=MAINTENANCE_COMPLETE;
+                        notify_status=MAINTENANCE_COMPLETE;
```
Comment (mvalas827) [inline MaintenanceManager/MaintenanceManager.cpp:556]: You can update m_notify_status (class variable) here itself after holding on to status mutex.

File: MaintenanceManager/MaintenanceManager.cpp [hunk 4]
```diff
@@ -561,47 +564,31 @@ namespace WPEFramework {
...
-                            /* even though we end up in skipped task /error
-                             * check if we have the reboot required is recevied */
-                            requestSystemReboot();
-                        }
-                        else {
-                            LOGINFO("Reboot Not Required !!");
-                        }
                     }
 
+                    MaintenanceManager::_instance->onMaintenanceStatusChange(notify_status);
```
Comment (mvalas827) [inline MaintenanceManager/MaintenanceManager.cpp:583]: move notification after thread join.

File: MaintenanceManager/MaintenanceManager.cpp [hunk 5]
```diff
@@ -831,20 +820,28 @@ namespace WPEFramework {
...
+                    m_statusMutex.unlock();
                     returnResponse(result);
                 }
 
         void MaintenanceManager::onMaintenanceStatusChange(Maint_notify_status_t status) {
             JsonObject params;
             /* we store the updated value as well */
-            g_notify_status=status;
+            m_statusMutex.lock();
+            m_notify_status=status;
```
Comment (mvalas827) [inline MaintenanceManager/MaintenanceManager.cpp:843]: As commented before, you can remove this status update here as the class variable is updated and controlled by mutex.

File: MaintenanceManager/MaintenanceManager.cpp [hunk 6]
```diff
@@ -374,6 +374,11 @@ namespace WPEFramework {
             /* we moved every thing to a thread */
             /* only when dcm is getting a DCM_SUCCESS/DCM_ERROR we say
              * Maintenance is started until then we say MAITENANCE_IDLE */
+
```
Comment (mvalas827) [inline MaintenanceManager/MaintenanceManager.cpp:377]: I don't see a need for this. If the global/class variables are properly protected and notifications are sent appropriately.

File: MaintenanceManager/MaintenanceManager.cpp [hunk 7]
```diff
@@ -831,20 +820,28 @@ namespace WPEFramework {
                         /* we set this to false */
                         g_is_critical_maintenance="false";
 
+
+                        if(m_thread.joinable()){
```
Comment (mvalas827) [inline MaintenanceManager/MaintenanceManager.cpp:824]: I don't see a need for this. If the global/class variables are properly protected and notifications are sent appropriately.

File: MaintenanceManager/MaintenanceManager.cpp (hunk 1 above)
Comment (mvalas827) [reply MaintenanceManager/MaintenanceManager.cpp:804]: Seems fine to me as the start method will return with in progress error. If start method was able to acquire lock then the previous execution should have been completed. Let me know if you still see some race condition here, we can discuss.

File: MaintenanceManager/MaintenanceManager.cpp [hunk 8]
```diff
@@ -561,47 +564,31 @@ namespace WPEFramework {
...
-                            LOGINFO("Reboot Not Required !!");
-                        }
                     }
 
+                    MaintenanceManager::_instance->onMaintenanceStatusChange(notify_status);
+                    LOGINFO("ENDING MAINTENANCE CYCLE"); 
                     if(m_thread.joinable()){
                         m_thread.join();
                     }
                 }
```
Comment (mvalas827) [inline MaintenanceManager/MaintenanceManager.cpp:588]: You can release mutex after thread join

**Review by mvalas827 — CHANGES_REQUESTED** [review]

**Review by anand-ky — APPROVED** [review]

**Pull request merged**